src/dbnode/storage/types.go

```diff
@@ -282,7 +291,15 @@ type databaseNamespace interface {
 		ctx context.Context,
 		query index.Query,
 		opts index.QueryOptions,
-	) (index.QueryResults, error)
+	) (index.QueryReturnResults, error)
+
+	// AggregateQuery resolves the given query into aggregated tags.
+	AggregateQuery(
+		ctx context.Context,
+		query index.Query,
+		opts index.QueryOptions,
+		aggResultOpts *index.AggregateResultsOptions,
+	) (index.QueryReturnResults, error)
```

Comment:
Why don't we actually just make methods for every single type of query?  Then you can concretely return `index.AggregateResults` for `AggregateQuery(...)`.

Inside the `storage/index.go` file you can do the magic to make both of the methods actually use the same underlying code, that way callers of these methods don't need to cast `index.Results` into a concrete type (of either `index.QueryResults` or `index.AggregateResults`).